### PR TITLE
Add assertion to check if the order of test paths is correct

### PR DESCRIPTION
**Pull request by ono-max**
Currently, we do not verify the order of JSON at https://github.com/launchableinc/cli/blob/main/tests/cli_test_case.py#L221 because we evaluate JSON in an order-agnostic manner. However, we aim to ensure the order of test paths is accurate. `assert_json_orderless_equal` disregards the order of items intentionally. Consequently, I have introduced the `assert_test_path_orderly_equal` method to address this requirement, instead of modifying `assert_json_orderless_equal`.

@kohsuke 

I think you designed these methods in https://github.com/launchableinc/cli/pull/91. Can you take a look at this PR?

### CODE REVIEW

File: tests/cli_test_case.py [hunk 1]
```diff
@@ -294,3 +295,15 @@ def tree_sorted(obj):
                 return obj
 
         self.assertEqual(tree_sorted(a), tree_sorted(b))
+
+    def assert_test_path_orderly_equal(self, a, b):
```
Comment (kohsuke) [inline tests/cli_test_case.py:299]: Let's define the contract of this method. What exactly does this method assert?

I ask because I was expecting `a` and `b` are symmetric, but L308-309 doesn't work that way. For exampe, if `a` is empty list, this assertion passes no matter what `b` is.

I'm almost tempted to just test `a['events']['testPath']==b['events']['testPath']`; Is that not what we want to do?

File: tests/test_runners/test_minitest.py [hunk 2]
```diff
@@ -20,27 +20,6 @@ def test_record_test_minitest(self):
         self.assert_success(result)
         self.assert_record_tests_payload('record_test_result.json')
 
-    # For testing https://github.com/launchableinc/cli/pull/846/files#r1591707246.
```
Comment (kohsuke) [inline tests/test_runners/test_minitest.py:23]: Is this removal intentional?

File: tests/cli_test_case.py (hunk 1 above)
Comment (ono-max) [reply tests/cli_test_case.py]: > I'm almost tempted to just test a['events']['testPath']==b['events']['testPath']; Is that not what we want to do?

I initially planed to do, but it's not possible because the value associated with the 'events' key is in random order.

**Comment by kohsuke** [comment]
Thanks! There's some loose ends when duplicates are present but now the intent is clar, so that's OK for now.
